src/dnd-list/stateful-list-container.js

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import * as React from 'react';
+import type {
+  StateT,
+  StatefulComponentContainerPropsT,
+  StateChangeTypeT,
+  StateReducerT,
+} from './types.js';
+import {arrayMove, arrayRemove} from 'react-movable';
+
+const defaultStateReducer: StateReducerT = (type, nextState) => nextState;
+
+class StatefulListContainer extends React.Component<
+  StatefulComponentContainerPropsT,
+  StateT,
+> {
+  static defaultProps: $Shape<StatefulComponentContainerPropsT> = {
+    initialState: {items: []},
+    stateReducer: defaultStateReducer,
+  };
+
+  state = {
+    items: [],
+    ...this.props.initialState,
+  };
+
+  onChange = ({oldIndex, newIndex}: {oldIndex: number, newIndex: number}) => {
+    const newItemsState =
+      newIndex === -1
+        ? arrayRemove(this.state.items, oldIndex)
+        : arrayMove(this.state.items, oldIndex, newIndex);
+    if (typeof this.props.onChange === 'function') {
```

Comment:
`this.props.onChange` is an optional prop and can be `undefined`. Since it's a stateful component, the onChange prop here serves more of informative purposes (compared to the stateless version when you have to use it to manipulate the items).

Reply:
We could set it to a noop function in the `defaultProps` and skip the check here. But flow make still need this check.